pilot/cmd/pilot-agent/status/server_test.go

```diff
@@ -217,3 +222,78 @@ func TestHttpsAppProbe(t *testing.T) {
 		}
 	}
 }
+
+func TestHandleQuit(t *testing.T) {
+	statusPort := 15020
+	s, err := NewServer(Config{StatusPort: uint16(statusPort)})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name       string
+		method     string
+		remoteAddr string
+		expected   int
+	}{
+		{
+			name:       "should send a sigterm for valid requests",
+			method:     "POST",
+			remoteAddr: "127.0.0.1",
+			expected:   http.StatusOK,
+		},
+		{
+			name:       "should send a sigterm for valid ipv6 requests",
+			method:     "POST",
+			remoteAddr: "[::1]",
+			expected:   http.StatusOK,
+		},
+		{
+			name:       "should require POST method",
+			method:     "GET",
+			remoteAddr: "127.0.0.1",
+			expected:   http.StatusMethodNotAllowed,
+		},
+		{
+			name:     "should require localhost",
+			method:   "POST",
+			expected: http.StatusForbidden,
+		},
+	}
+	for _, scope := range log.Scopes() {
```

Comment:
Why do we need this in the test?

Reply:
Oops, good catch. I was debugging something